src/twisted/conch/test/test_ckeygen.py

```diff
@@ -628,3 +629,59 @@ def test_changePassphraseSubtypeV1(self):
         self.assertTrue(
             privateKeyContent.startswith(b"-----BEGIN OPENSSH PRIVATE KEY-----\n")
         )
+
+    def test_useDefaultForKey(self):
+        """
+        L{options} will default to "~/.ssh/id_rsa" if the user doesn't
+        specify a key.
+        """
+
+        def mock_input(*args):
+            return ""
+
+        import builtins
+
+        self.patch(builtins, "input", mock_input)
+        options = {"filename": ""}
+        _getKeyOrDefault(options)
+        self.assertNotEqual(
+            options["filename"],
+            "",
+        )
+        self.assertTrue(
+            "id_rsa" in options["filename"],
+        )
+
+    def test_displayPublicKeyHandleFileNotFound(self):
+        """
+        Ensure FileNotFoundError is handled, whether the user has supplied
+        a bad path, or has no key at the default path.
+        """
+        options = {"filename": "/foo/bar"}
+        with self.assertRaises(SystemExit) as e:
+            displayPublicKey(options)
+            self.assertTrue(
+                "could not be opened, please specify a file." in e,
+            )
+
+    def test_changePassPhraseHandleFileNotFound(self):
+        """
+        Ensure FileNotFoundError is handled for an invalid filename.
+        """
+        options = {"filename": "/foo/bar"}
+        with self.assertRaises(SystemExit) as e:
+            changePassPhrase(options)
+            self.assertTrue(
+                "could not be opened, please specify a file." in e,
+            )
+
+    def test_printFingerprintHandleFileNotFound(self):
+        """
+        Ensure FileNotFoundError is handled for an invalid filename.
+        """
+        options = {"filename": "/foo/bar", "format": "md5-hex"}
+        with self.assertRaises(SystemExit) as e:
+            printFingerprint(options)
+            self.assertTrue(
+                "could not be opened, please specify a file." in e,
+            )
```

Comment:
There is a lot of repetition in these tests... but this is just a minor comment.